Journey formatting and Deploy to Bluemix

markstur opened the pull request:
Make the top of README and quick start look like a journey.
Add Deploy to Bluemix button running with polymer serve for now.
Lower README and Tutorial is left as-is, but the flow from journey to the rest is not complete.

markstur committed:
Add a manifest.yml and deps for cloud foundry build/run

Added dependencies so that the app will build/run with a cf push.
Currently using polymer build/es5-bundled.
Running with polymer serve for now.

markstur committed:
Add test deploy to bluemix button

markstur committed:
README work to make it look like a journey

markstur committed:
Add images for README

markstur committed:
Move tutorial to a subdirectory like other impls.

Moving the tutorial to a subdir significantly cleans up
the main README.md. In particular this seems like a good
strategy for making this a journey. Also this is following
the existing practice of some of the other Offline First
shopping list implementations.

markstur committed:
Add the usual ACK, CONTRIB, MAINT files for journeys.

markstur committed:
Fix eslint errors

bradnoble commented:
I heard this am that "Journeys" have been renamed to "Patterns". @markstur true?

markstur commented:
Yes. s/journey/pattern/ 
OK, I attempted to use Code Pattern.  This is a new thing and most repos say journey, but I guess we'll use Code Pattern going forward.

markstur committed:
Fix typo in eslint code ~fix.

Previous commit fixing eslint errors introduced a typo
that shows up when testing with replication.

markstur committed:
Remove Features details...

Remove the details of the Features epic and just link to the
master list for shopping list features. Mention features that are
not yet implemented.

Expand on "Using the app" a little bit more. The app is easy
enough to use w/o instructions, but the README should explain
enough to give a good idea of what it does.

markstur committed:
Try a better screen shot of shopping lists

markstur committed:
Add the Links and License section. Tweak some headers.

markstur committed:
Use express.static server and add metrics tracker.

Use express.static server instead of polymer serve.
Add metrics tracker.
Update deploy button and add badge.

bradley-holt commented on CONTRIBUTING.md:
```diff
@@ -0,0 +1,59 @@
+feel free to scratch it.
+
+To contribute code or documentation, please submit a pull request to the [GitHub
+repository](https://github.com/IBM/shopping-list-polymer-pouchdb).
```
The repo link here is incorrect.

markstur replied:
I changed all the links from IBM to ibm-watson-data-lab

bradley-holt commented on CONTRIBUTING.md:
```diff
@@ -0,0 +1,59 @@
+
+A good way to familiarize yourself with the codebase and contribution process is
+to look for and tackle low-hanging fruit in the [issue
+tracker](https://github.com/IBM/shopping-list-polymer-pouchdb/issues). Before embarking on
```
Incorrect repo link again.

bradley-holt replied:
Looks like there's a few more of these in here…

bradley-holt commented on CONTRIBUTING.md:
```diff
@@ -0,0 +1,59 @@
+
+Please feel free to connect with us: [here](https://github.com/IBM/shopping-list-polymer-pouchdb/issues)
+
+## Setup
```
It appears that the "setup," "testing," and "coding style guidelines" sections are still placeholders?

markstur replied:
We have bad habit of leaving this file as-is, but fixed it now. Mostly refers back to README.

bradley-holt commented on MAINTAINERS.md:
```diff
@@ -0,0 +1,74 @@
+
+## Methodology:
+
+A master branch. This branch MUST be releasable at all times. Commits and
```
This is different from how we have maintained this repository up until now, though I'm fine changing how we do this.

markstur replied:
We don't follow that either. I deleted the section and discussing with team. Normal open source github practices should be followed.

bradley-holt commented on README.md:
```diff
@@ -1,12 +1,61 @@
+## Featured technologies
+* [Polymer](https://www.polymer-project.org/): Libraries, tools, and patterns for building Progressive Web Apps using web platform features such as Web Components, Service Workers, and HTTP/2.
+* [PouchDB](https://pouchdb.com/): An open source JavaScript database that syncs with anything that implements the CouchDB Replication Protocol.
+* [Databases](https://en.wikipedia.org/wiki/IBM_Information_Management_System#.22Full_Function.22_databases): Repository for storing and managing collections of data.
```
I'm not sure what this part means, exactly.

markstur replied:
If you mean specifically the desc of "Databases". I agree. That's our standard blurb from. Otherwise we use the components list for IBM stuff and featured tech for other stuff. So I like to use this to at least call out Polymer and PouchDB. I ended up moving other stuff to "Links" at the bottom.

bradley-holt commented on README.md:
```diff
@@ -20,13 +69,73 @@ Third, install the [Bower npm resolver](https://github.com/mjeanroy/bower-npm-re
+
+![](doc/source/images/shopping_lists.png)
+
+## Epic features
```
Perhaps "Features as epics" might make more sense as the heading here (or just "Features")?

markstur committed:
Address feedback.

* Try to say patterns instead of journey in README
* Customize/fix some of the usual templated md files.
* Add git clone step for run locally.

markstur committed:
One more s/journey/pattern/

markstur committed:
Use build/es5-bundled in app.js with npm start (Bluemix)

Running out of the es5-bundled build. Hardcoded in app.js
to make the default most compatible with browsers.

markstur committed:
Add Privacy Notice and how to disable the tracker

Need to add this notice to the README because we use
tracker code.

bradley-holt reviewed (approved):
LGTM
